### DISCUSSION TITLE
Client Credentials Grant

### DISCUSSION THREAD

**Discussion by swissmilo** [panva/openid-client#318] [Q&A] [answered]
Hi there, I wanted to ask about using Client Credential Grant types. I couldn't see a way to change the Grant Type from Authorization Code when creating the Client object. Is the correct way of obtaining a set of tokens using a Client Credentials Grant something like this? Much appreciate a pointer in the right direction :)

`const issuer = await Issuer.discover('https://.../openid-configuration');

const client = new issuer.Client({
    client_id: 'XYZ',
    client_secret: '123',
  }); 

  const tokenSet = client.grant({
    grant_type: 'client_credentials'
  })

  console.log('validated Token claims %s', tokenSet.claims);
`

**Comment by panva** [comment, marked as answer]
> Is the correct way of obtaining a set of tokens using a Client Credentials Grant something like this?

Yeah, altho its worth mentioning the only "validation" in place is TLS. The response is not processed in any way.
